hystrix-contrib/hystrix-metrics-event-stream/src/main/java/com/netflix/hystrix/contrib/metrics/eventstream/HystrixMetricsStreamServlet.java

```diff
@@ -135,7 +134,17 @@ private void handleRequest(HttpServletRequest request, HttpServletResponse respo
                 response.setHeader("Cache-Control", "no-cache, no-store, max-age=0, must-revalidate");
                 response.setHeader("Pragma", "no-cache");
 
-                MetricJsonListener jsonListener = new MetricJsonListener();
+                int queueSize = defaultMetricListenerQueueSize.get();
+                try {
+                    String q = request.getParameter("queueSize");
```

Comment:
As I mention in my previous comment, there doesn't seem to be a way of scaling the queue size if the number of backend endpoints is scaled up or down.

For instance, a user-facing email service may only need 25 instances of a backend email retrieval service during off-peak but auto-scale to 100-150 during peak periods. In such a case, we would need to estimate the peak load, and set the queue size accordingly. If the volume of metrics exceeded that, the hystrix stream servlet would need to be reconfigured each time.

The consumer of metrics would be in a better position to anticipate the typical patterns in load, and adjust accordingly if it noticed the stats collection to be failing due to hitting queue size caps.